packages/cosmos/src/client.rs

```diff
@@ -147,6 +148,74 @@ impl<Res> PerformQueryWrapper<Res> {
 }
 
 impl Cosmos {
+    async fn get_and_update_simulation_sequence(
+        &self,
+        address: Address,
+    ) -> Result<BaseAccount, Error> {
+        let mut guard = self.pool.get().await?;
+        let cosmos = guard.get_inner_mut();
+        let mut sequences = cosmos.simulate_sequences.write().await;
```

Comment:
This approach means we're grabbing a write lock and holding onto it while querying the chain, which will likely block other tasks fairly often. It would be best to avoid that.